user.c

```diff
@@ -0,0 +1,228 @@
+    int userAge;
+} Person;
+
+int checkUserID(int id)
+{
+    int result = 0; // Default result: User ID not found
+    FILE *fp = fopen("users_data.txt", "r");
+    if (!fp)
+    {
+        result = -1; // File error
+    }
+    else
+    {
+        Person tempUser;
+        while (fscanf(fp, "%d,%49[^,],%d", &tempUser.userID, tempUser.userName, &tempUser.userAge) == 3)
+        {
+            if (tempUser.userID == id)
+            {
+                result = 1; // User ID exists
+                break;
+            }
+        }
+        fclose(fp);
+    }
+    return result;
+}
+
+void addUser()
+{
+    Person newUser;
+    printf("Enter User ID: ");
+    scanf("%d", &newUser.userID);
+    if (checkUserID(newUser.userID) == 1)
+    {
+        printf("Error: User ID %d already exists\n", newUser.userID);
+        return;
+    }
+    printf("Enter Username: ");
+    scanf(" %49s", newUser.userName);
+    printf("Enter User Age: ");
+    scanf("%d", &newUser.userAge);
+
+    FILE *fp = fopen("users_data.txt", "a");
+    if (!fp)
+    {
+        printf("Error opening file\n");
+        return;
+    }
+    fprintf(fp, "%d,%s,%d\n", newUser.userID, newUser.userName, newUser.userAge);
+    fclose(fp);
+    printf("User added successfully\n");
+}
+
+void listUsers()
+{
+    Person tempUser;
+    FILE *fp = fopen("users_data.txt", "r");
+    if (!fp)
+    {
+        printf("Unable to open file\n");
+        return;
+    }
+    printf("UserID  Username  Age\n\n");
+    while (fscanf(fp, "%d,%49[^,],%d", &tempUser.userID, tempUser.userName, &tempUser.userAge) == 3)
+    {
+        printf("   %d    %s    %d\n", tempUser.userID, tempUser.userName, tempUser.userAge);
+    }
+    fclose(fp);
+}
+
+void modifyUser()
+{
+    int idToUpdate, found = 0;
+    Person tempUser;
+    FILE *fp = fopen("users_data.txt", "r");
+    if (!fp)
+    {
+        printf("Error opening file\n");
+        return;
+    }
+
+    Person userList[100];
+    int totalUsers = 0;
+
+    while (fscanf(fp, "%d,%49[^,],%d", &tempUser.userID, tempUser.userName, &tempUser.userAge) == 3)
+    {
+        userList[totalUsers++] = tempUser;
+    }
+    fclose(fp);
+
+    printf("Enter User ID to update: ");
+    scanf("%d", &idToUpdate);
+    for (int i = 0; i < totalUsers; i++)
+    {
+        if (userList[i].userID == idToUpdate)
+        {
+            found = 1;
+            printf("Enter new Username: ");
+            scanf(" %49s", userList[i].userName);
+            printf("Enter new User Age: ");
+            scanf("%d", &userList[i].userAge);
+            printf("\nUpdated successfully:\n");
+            printf("UserID: %d, Username: %s, Age: %d\n", userList[i].userID, userList[i].userName, userList[i].userAge);
+            break;
+        }
+    }
+    if (!found)
+    {
+        printf("Error: User ID %d not found\n", idToUpdate);
+        return;
+    }
+    fp = fopen("users_data.txt", "w");
+    if (!fp)
+    {
+        printf("Error opening file\n");
+        return;

```

Comment:
Why same things are repeated here? I have added comment earlier that the shouldn't be multiple return statement in single function.